st2api/st2api/controllers/resource.py

```diff
@@ -211,6 +211,15 @@ def _get_one_by_id(self, id, requester_user, permission_type, exclude_fields=Non
 
         instance = self._get_by_id(resource_id=id, exclude_fields=exclude_fields)
 
+        # A bit gross, but it seems necessary at the moment in order to
+        # make RBAC happy, since Inquiries don't currently have their own
+        # DB model
+        # TODO remove this atrocity once InquiryDB is a thing
```

Comment:
Please explain how this make RBAC happy.

Reply:
As with most strange things I had to do in with RBAC for Inquiries, it all comes down to the fact that Inquiries don't have their own DB model. I also would like to say this is pretty much the only place where that fact made things difficult. For the most part, not having an Inquiries DB model didn't really affect my work. But in RBAC it reared its head a few times.

When I run `_get_one` from the Inquiries controller, the functions in the `Resource` base class pull a resource [directly off of the database by resource id](https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/resource.py#L268) **and then** does its own RBAC assertions all before getting back to the Inquiries controller. So without intervention, the resource that makes it into RBAC is of the format `execution:<id>`. That's when my tests start failing:

```
======================================================================
FAIL: test_user_has_resource_db_permission (tests.unit.test_rbac_resolvers_inquiry.InquiryPermissionsResolverTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/st2/st2common/tests/unit/test_rbac_resolvers_inquiry.py", line 256, in test_user_has_resource_db_permission
    permission_type=PermissionType.INQUIRY_VIEW)
  File "/home/vagrant/st2/st2common/tests/unit/test_rbac_resolvers.py", line 105, in assertUserHasResourceDbPermission
    raise AssertionError(msg)
AssertionError: Expected permission grant "inquiry_view" for user "custom_role_inquiry_view_grant" on resource DB "execution:59caf55032ed3503f189573a", but no grant was found
```

Since all of that is done before even getting back to the Inquiries controller code, the only option I have is to intercept that resource right before the assertion and make this admittedly duct-tape-esque conditional, replacing the "get_uid" function with this one, so that RBAC calls the correct resolver.

Hopefully that makes sense, and I will certainly improve the comment with some of this detail. And if you know of a better way, do let me know. I realize this is a bit icky.

Reply:
@Kami This implementation here don't look right. I need to review the RBAC code for this. Have you looked at this?

Reply:
It looks like you could perhaps re-implement `_get_one_by_id` in your controller class and that way we can avoid this "hack" here?

Would that work?

Reply:
That works. I changed this in 5596bfa9e, so the scope of the hack is limited. I also improved the comment so it was more obvious why I went this way